pkgs/development/libraries/ngtcp2/gnutls.nix

```diff
@@ -0,0 +1,51 @@
+{ lib, stdenv, fetchFromGitHub
+, autoreconfHook, pkg-config
+, gnutls
+, cunit, ncurses
+}:
+
+stdenv.mkDerivation rec {
+  pname = "ngtcp2";
+  version = "0.7.0";
+
+  src = fetchFromGitHub {
+    owner = "ngtcp2";
+    repo = "ngtcp2";
+    rev = "v${version}";
+    sha256 = "sha256-REAN5TW0miWXI3HFxtW3znTKTrhsBbNqu1VfjC2w0no=";
+  };
+
+  outputs = [ "out" "dev" ];
+
+  nativeBuildInputs = [ autoreconfHook pkg-config ];
+  buildInputs = [ gnutls ];
+
+  configureFlags = [ "--with-gnutls=yes" ];
+  enableParallelBuilding = true;
+
+  doCheck = true;
+  checkInputs = [ cunit ]
+    ++ lib.optional stdenv.isDarwin ncurses;
+
+  meta = with lib; {
+    homepage = "https://github.com/ngtcp2/ngtcp2";
+    description = "an effort to implement RFC9000 QUIC protocol.";
+    license = licenses.mit;
+    platforms = platforms.unix;
+    maintainers = with maintainers; [ vcunat/* for knot-dns */ ];
+  };
+}
+
+/*
+  Why split from ./default.nix?
+
+  ngtcp2 libs contain helpers to plug into various crypto libs (gnutls, patched openssl, ...).
+  Building multiple of them while keeping closures separable would be relatively complicated.
```

Comment:
Would it be a problem to compile openssl and gnutls into one package?

Reply:
You would need to put each of these into a separate output (and leaving the base lib in a third one) &ndash; and then patch paths in pkg-config and libtool files.

Reply:
No, that's a different one.  The openssl fork takes a few megabytes and gnutls also (and each additional crypto lib that you'd add).  Packages always want just one but they'd get all of them.

Reply:
The potential blowup in this comment is for the situation that got merged.  That is you maybe need (through different packages) ngtcp2 with both openssl and gnutls, in which case you get the core twice.  And the core is 0.3--0.4 MB.